geodata/models/geometry/base.py

```diff
@@ -0,0 +1,51 @@
+from django.contrib.gis.db import models
+from django.db import transaction
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from ..common import ModifiableEntry, SpatialEntry
+from ..constants import DB_SRID
+from ..mixins import PostSaveEventMixin
+from ... import tasks
+
+
+def validate_zip_extension(value):
+    import os
+    from django.core.exceptions import ValidationError
+
+    ext = os.path.splitext(value.name)[1]  # [0] returns path+filename
+    valid_extensions = [
```

Comment:
Instead of using extension, we could also validate via file contents (using the python magic library, for instance, to ensure it is a zip file).

Is there something else that takes advantage of this being a zip file, or could we more broadly also accept tar files?

Reply:
It could be any type of archive - zip/tar/whatever. I just chose zip for simplicity.

I will update this yo use the python magic library to verify the archive and also accept tar

Reply:
I updated this to check the mime type. @manthey, can you take a look to see if this is what we want?